mesonbuild/backend/backends.py

```diff
@@ -350,7 +350,7 @@ def get_target_filename_for_linking(self, target: T.Union[build.Target, build.Cu
         if isinstance(target, build.SharedLibrary):
             link_lib = target.get_import_filename() or target.get_filename()
             # In AIX, if we archive .so, the blibpath must link to archived shared library otherwise to the .so file.
-            if mesonlib.is_aix() and target.aix_so_archive:
+            if mesonlib.Platform.is_aix and target.aix_so_archive:
```

Comment:
This is sorta tangental, but I'm not sure the use of `meson.is*` is actually correct here, I think this needs to be `Environment.machines[target.for_machine].is_aix`

Reply:
Maybe cross-compiling to AIX is less common than cross compiling between Windows and Linux. :D

Reply:
Absolutely true :D